### PR TITLE
Elia/add swagger

### PR DESCRIPTION
**Pull request by eliakaris**
https://damienbod.com/2015/12/13/asp-net-5-mvc-6-api-documentation-using-swagger/

Adding swagger to the server.

### CODE REVIEW

File: Website/Src/Startup.Container.cs [hunk 1]
```diff
@@ -187,6 +188,21 @@ public void ConfigureServices(IServiceCollection services)
             // Allow IOptions<T> to be available through DI
             services.AddOptions();
 
+            services.AddSwaggerGen(options =>
+            {
+                options.SwaggerDoc(
+                    "v1",
```
Comment (dfederm) [inline Website/Src/Startup.Container.cs]: Is this required? At least for now we don't really have a versioning story (or plan to)

File: Website/Src/Startup.Container.cs [hunk 2]
```diff
@@ -187,6 +188,21 @@ public void ConfigureServices(IServiceCollection services)
...
+                    new Info
+                    {
+                        Title = "ClickerHeroesTracker API",
+                        Version = "v1",
```
Comment (dfederm) [inline Website/Src/Startup.Container.cs]: Same

File: Website/Src/Startup.Container.cs [hunk 3]
```diff
@@ -187,6 +188,21 @@ public void ConfigureServices(IServiceCollection services)
...
+                        Description = "Api for Clicker Heroes",
```
Comment (dfederm) [inline Website/Src/Startup.Container.cs]: Technically an API for Clicker Heroes Tracker, not Clicker Heroes :)

File: Website/Src/Startup.cs [hunk 4]
```diff
@@ -59,6 +59,16 @@ public void Configure(IApplicationBuilder app)
                     defaults: new { controller = "Home", action = "Beta" });
             });
 
+            app.UseSwagger(c =>
+            {
+                c.PreSerializeFilters.Add((swagger, httpReq) => swagger.Host = httpReq.Host.Value);
```
Comment (dfederm) [inline Website/Src/Startup.cs]: Can you add a comment for what this line does? For me at least it's not very clear.

Comment (eliakaris) [reply Website/Src/Startup.cs]: Sure, this was mainly copy / paste.  Let me see what all I can remove.  Also looking into build time generation of this.

File: Website/Src/Startup.Container.cs (hunk 1 above)
Comment (eliakaris) [reply Website/Src/Startup.Container.cs]: Not required, let me check up on all the options of this to have what we want.

**Comment by eliakaris** [comment]
Not sure why I cant reply to your comment...  I will have a separate change which builds the swagger.json file as part of the build.  Then on the other side of the build, the client can gen the service files from this json.

This will require the client app has a build dependency on the server app.  I'm guessing this is OK.

**Pull request closed without merging**